jerry-core/jerryscript.h

```diff
@@ -389,6 +389,11 @@ size_t jerry_parse_and_save_literals (const jerry_char_t *source_p, size_t sourc
                                       uint32_t *buffer_p, size_t buffer_size, bool is_c_format);
 
 /**
+ * External context functions.
+ */
+size_t jerry_get_context_size (void);
+size_t jerry_get_lcache_size (size_t row_count, size_t row_length);
```

Comment:
> My problem is we export internals to users, and these internals can change, while public API should not.

Hmm this is kind of the opposite of what you suggested [earlier]... ;P

I think using a [config struct] + making `jerry_instance_t` a fwd declaration helps with future API compatibility (at least semantically / "do projects still compile"):

If the internals would be removed -- say, just for the sake of discussion: LCache would be removed (unlikely scenario, I know): the `lcache` struct field would be marked deprecated, but left in place (so projects still compile, but with a warning) and its values would be ignored at run time.

If the internals would be extended -- say some fancy new regex cache is added: new config fields are added to the `jerry_instance_config_t` struct. For existing projects using the API, the values would be zero and thus defaults would be used.

> Furthermore if I understand correctly, the problem with jerry_instance_create that it cannot allocate memory (since jerry does not depend on malloc).

No, you don't need malloc to implement that `alloc` callback. The simplest callback would select a statically allocated buffer and return it.

Re. `jerry_get_instance_size`: you would still have to expose the internals of `jerry_instance_t`, so if the internals change, you'd have to change that struct and break existing projects. In [my proposal], that struct is internal / hidden / forward declared.

[earlier]: https://github.com/jerryscript-project/jerryscript/pull/1778/files#r113156069
[my proposal]: https://github.com/jerryscript-project/jerryscript/pull/1778#discussion_r114267745
[config struct]: https://github.com/jerryscript-project/jerryscript/pull/1778#discussion_r114267745